scalableminds/webknossos
# Store editable mappings in multiple fossildb columns+keys

### URL of deployed dev instance (used for testing):
- https://editablemappingsdistributed.webknossos.xyz

### TODO
 - [x] on multiple update actions, use temporary store, flush to fossil on commit
 - [x] problem: newest version may not be materialized, do we apply update actions in migration? or should the code be able to apply lazily?
 - [x] test perf
 - [x] how to apply migration in production? how much downtime is needed?
 - [x] two-phase migration to lower downtime?
 - [x] include relevant changes from #6829
 - [x] finalize migration (args, camel_case)
 - [ ] Reproduce philipps failed mesh requests → cannot reproduce, let’s defer.
 - [x] Front-end should not send fallback layer requests if volume tracing bucket requests fail
 - [x] should action with segment id 0 also become no-op? yes

### Steps to test:
- Create annotation with volume layer with segmentation fallback (that has an agglomerate file. on the dev instance I suggest `l4dense_motta_et_al_dev_v2`)
- Select an agglomerate mapping and use the proofreading tool to split and merge some cells
- refresh the page, should still be look right
- To test the migration locally, repeat above steps on master, then switch to this branch, start the fossildb (yarn start does it), then run the migration `python tools/migrate-editable-mappings/migrate-editable-mappings.py -vw localhost:7155` , then reload the same annotation and see that everything is there and still works as expected

### Issues:
- fixes #6797

------
